src/Sulu/Bundle/MediaBundle/Content/Types/SingleMediaSelection.php

```diff
@@ -186,4 +190,24 @@ public function mapPropertyMetadata(ContentPropertyMetadata $propertyMetadata):
 
         return new PropertyMetadata($propertyMetadata->getName(), $mandatory, $singleMediaSelectionMetadata);
     }
+
+    public function getReferences(PropertyInterface $property, PropertyValue $propertyValue, ReferenceCollector $referenceCollector): void
+    {
+        $data = $propertyValue->getValue();
+        if (!isset($data['id'])) {
+            return;
+        }
+
+        $media = $this->mediaManager->getById($data['id'], $referenceCollector->getReferenceLocale());
+
+        $referenceCollector->addReference(
+            MediaInterface::RESOURCE_KEY,
+            $data['id'],
+            $media->getName(),
```

Comment:
I think the `media->getName()` is not required here as the media list should show the pages title and the pages title should be written on the collector already.